### PR TITLE
[ShadowContainer] CornerRadius performance issue

### LINKED ISSUE #789: [ShadowContainer] CornerRadius performance issue
## What would you like to be added:

The PR [[ShadowContainer] Inner Shadows not properly fitted in Rectangle and Ellipse](https://github.com/unoplatform/uno.toolkit.ui/pull/757#top) add the CornerRadius handle for non uniform corner radius.

But we need to update the performance of the code, so we open this issue referring to the discussion in the original PR
https://github.com/unoplatform/uno.toolkit.ui/pull/757#discussion_r1310738482

So now we need to change the usage of linq to array.

## Why is this needed:

For better performance.

## For which Platform:

- [x] WebAssembly
- [ ] Android
- [ ] iOS
- [ ] macOS (AppKit)
- [ ] Mac Catalyst
- [x] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Linux Framebuffer
  - [ ] Tizen
- [ ] Windows

### PR DESCRIPTION
**Pull request by rafael-rosa-knowcode**
GitHub Issue (If applicable): closes: #789 

## PR Type

What kind of change does this PR introduce?

- Refactoring (no functional changes, no api changes)

## What is the current behavior?

For better performance.

## What is the new behavior?

Using array.

## PR Checklist

Please check if your PR fulfills the following requirements:
- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Tested the changes where applicable:
	- [ ] UWP
	- [ ] WinUI
	- [ ] iOS
	- [ ] Android
	- [ ] WASM
	- [ ] MacOS
- [ ] Updated the documentation as needed:
	- [ ] [General Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc)
	- [ ] [Controls Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc/controls)
	- [ ] [Extensions Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc/helpers)
	- [ ] [controls-styles.md](https://github.com/unoplatform/uno.toolkit.ui/blob/main/doc/controls-styles.md)
	- [ ] [lightweight-styling.md (LightWeight Styling Resource Keys)](https://github.com/unoplatform/uno.toolkit.ui/blob/main/doc/lightweight-styling.md)
- [ ] [Runtime Tests and/or UI Tests](https://platform.uno/docs/articles/contributing/guidelines/creating-tests.html) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal)
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.Paint.cs [hunk 1]
```diff
@@ -258,7 +258,7 @@ protected override SKRoundRect GetContentShape(ShadowPaintState state)
 		{
 			var rect = new SKRect(0, 0, (float)Width * state.PixelRatio, (float)Height * state.PixelRatio);
 			var radii = new SKPoint[] {
-				new SKPoint((float)(state.PixelRatio * CornerRadius.TopLeft), (float)(state.PixelRatio * CornerRadius.TopLeft)),
+				new SKPoint((float)CornerRadius.TopLeft * state.PixelRatio, (float)CornerRadius.TopLeft * state.PixelRatio),
```
Comment (jeromelaban) [inline src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.Paint.cs:261]: @rafael-rosa-knowcode I'd think the original comment was about all lines

File: src/Uno.Toolkit.RuntimeTests/Tests/ShadowContainerTests.cs [hunk 2]
```diff
@@ -28,13 +28,102 @@
...
+
+			//TopRight
+			await renderer.AssertColorAt(inner ? Colors.Red : Colors.Yellow, (int)((stackPanel?.ActualWidth ?? 0) - (inner ? absOffsetX + 1 : 1)), inner ? absOffsetY + 1 : 1);
+
+			//BottonLeft
+			await renderer.AssertColorAt(Colors.Yellow, 1, (int)((stackPanel?.ActualHeight ?? 0) - (inner ? absOffsetX + 1 : 1)));
+
+			//BottonRight and CornerCurve
+			var color = bottomRightCorner > 50 ? Colors.Yellow : (offsetX < 0 ? inner ? Colors.Red : Colors.Yellow : inner ? Colors.Green : Colors.Red);
+			await renderer.AssertColorAt(color, (int)((stackPanel?.ActualWidth ?? 0) -(inner ? absOffsetX + 1 : 1)), (int)((stackPanel?.ActualHeight ?? 0) - (inner ? absOffsetY + 1 : 1)));
```
[the remark this answers is not in the document]
Comment (rafael-rosa-knowcode) [reply src/Uno.Toolkit.RuntimeTests/Tests/ShadowContainerTests.cs]: Updated.

**Comment by rafael-rosa-knowcode** [comment]
During testing on wasm.
I noticed that the ImageAssertHelper.IsScreenshotSupported validation takes into account whether Microsoft.UI.Xaml.Media.Imaging.RenderTargetBitmap is implemented.
And it presents the false result.
However it works locally, even within the WinUI project.


When the validation is present the test shows Inconclusive.

image

When the validation is removed the test works fine on wasm.

image

In this case, we keep ImageAssertHelper.IsScreenshotSupported()?

**Review by agneszitte — APPROVED** [review]

**Comment by agneszitte** [comment]
@Mergifyio backport legacy/3x release/stable/4.2

**Pull request merged**